guides/k8s-microservices/src/main/paradox/index.md

```diff
@@ -0,0 +1,292 @@
+are part of a StatefulSet have a sticky, unique identity that helps bootstrap the Akka cluster. 
+To do this, the first node is used as seed node and referenced by environment variables. Chirper defines `StatefulSet`
+resources for each of its services: `friendservice`, `activityservice`, `chirpservice`, and `web`
+
+##### Service
+
+A [Service](https://kubernetes.io/docs/concepts/services-networking/service/) provides the means to expose TCP and
+UDP ports to other pods within the cluster.
+
+Chirper requires TCP ports to be exposed for Akka remoting. This is done using a `Service` declaration. The exposed 
+port will be used by `StatefulSet` declarations to bootstrap the Akka cluster.
+
+Chirper also exposes various HTTP endpoints from the microservices defined within the Chirper itself. The TCP port 
+for each service is also exposed using `Service` declaration. The name `http-lagom-api` is used as the name
+thereby allowing these endpoints to be looked up via Kubernetes DNS using DNS SRV lookup.
+
+##### Ingress
+
+An [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) is a collection of rules that 
+allow external traffic to reach services running inside Kubernetes. This allows, for example, 
+requests to `/api/users` to be routed to the `friendservice` while requests for `/` are routed to `web`. It also gives 
+you a central place to terminate TLS and Chirper is configured to do this for you.
+
+Chirper is configured to use nginx as the ingress controller.
+
+### Installation script
+
+_Below, the install script in the Chirper repository is dissected. You can find it in its
+entirety at `deploy/k8s/minikube/scripts/install`_
+
+Now that all the resources required for deployment have been described, this guide will cover how to automate the process
+of deploying them to Kubernetes.
+
+Deploying Chirper requires the following actions:
+
+* Start up your Minikube
+* Deploy Cassandra
+* Build Chirper Docker images
+* Deploy Chirper
+* Deploy nginx
+* Open the service in your browser
+
+Let's take a look at how these tasks can be scripted by analyzing `install`. Feel free to run these commands
+from your own terminal as well. Should you do that, make sure you've `cd`'d into the Chirper repository.
+
+#### Starting up your Minikube
+
+For simplicity's sake, it's nice to start from a fresh Minikube installation. The following resets the state of the local
+Minikube and configures the environment variables of your terminal session to point Docker to it.
+ 
+```bash
+(minikube delete || true) &>/dev/null && \
+minikube start --memory 8192 && \
+eval $(minikube docker-env)
+```
+
+#### Deploy Cassandra
+
+To deploy Cassandra to Kubernetes, the requisite resources must be created. The command below will create the resources, wait for
+Cassandra to start up, and show you its status.
+
+
+```bash
+kubectl create -f deploy/k8s/minikube/cassandra && \
+deploy/k8s/minikube/scripts/kubectl-wait-for-pods && \
```

Comment:
Perhaps a quick mention that `deploy/k8s/minikube/scripts/kubectl-wait-for-pods` waits for all pods to be ready?